silkworm/rpc/http/request_handler.cpp

```diff
@@ -34,9 +34,20 @@ constexpr std::size_t kStreamBufferSize{4096};
 Task<void> RequestHandler::handle(const std::string& content) {
     auto start = clock_time::now();
     std::string response;
-
     bool send_reply{true};
-    const auto request_json = nlohmann::json::parse(content);
+    nlohmann::json request_json;
+    bool parse_error = {false};
+    try {
+        request_json = nlohmann::json::parse(content);
+    } catch (const nlohmann::json::exception& e) {
+        SILK_ERROR << "Connection::do_read json_parse: " << e.what();
+        parse_error = true;
+    }
+    if (parse_error) {
+        response = make_json_error(0, -32600, "invalid request").dump() + "\n";
+        co_await channel_->write_rsp(response);
+        co_return;
+    }
```

Comment:
I suggest moving this `catch` clause down in order to surround lines up to 78. This way you can get rid of the `parse_error` variable, the `if` branch and the duplicated `write_rsp`: you just need to fill the `response` using `make_json_error` within the `catch` block and then rely on already existing lines 80-82 to have the response sent out.